File: app/src/main/java/info/maaskant/wouttest2/di/InstrumentationModule.java
/*
 * The MIT License
 *
 * Copyright (c) 2013-2016 reark project contributors
 *
 * https://github.com/reark/reark/graphs/contributors
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 * THE SOFTWARE.
 */
package info.maaskant.wouttest2.di;

import android.app.Application;
import android.content.Context;


import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import info.maaskant.wouttest2.utils.ApplicationInstrumentation;
import info.maaskant.wouttest2.utils.DebugApplicationInstrumentation;
import info.maaskant.wouttest2.utils.LeakCanaryTracing;
import info.maaskant.wouttest2.utils.LeakTracing;
import okhttp3.Interceptor;
import okhttp3.OkHttpClient;

@Module
public class InstrumentationModule {

    @Provides
    @Singleton
    public ApplicationInstrumentation providesInstrumentation(LeakTracing leakTracing) {
        return new DebugApplicationInstrumentation(leakTracing);
    }

    @Provides
    @Singleton
    public LeakTracing providesLeakTracing(Application application) {
        return new LeakCanaryTracing(application);
    }

}

File: app/src/main/java/info/maaskant/wmsnotes/android/app/di/InstrumentationModule.kt
//package info.maaskant.wmsnotes.android.app.di
//
//import android.app.Application
//import dagger.Module
//import dagger.Provides
//import info.maaskant.wmsnotes.android.app.instrumentation.ApplicationInstrumentation
//import info.maaskant.wmsnotes.android.app.instrumentation.DebugApplicationInstrumentation
//import info.maaskant.wmsnotes.android.app.instrumentation.LeakCanaryTracing
//import info.maaskant.wmsnotes.android.app.instrumentation.LeakTracing
//
//import javax.inject.Singleton
//
//@Module
//class InstrumentationModule {
//
//    @Provides
//    @Singleton
//    fun providesInstrumentation(leakTracing: LeakTracing): ApplicationInstrumentation {
//        return DebugApplicationInstrumentation(leakTracing)
//    }
//
//    @Provides
//    @Singleton
//    fun providesLeakTracing(application: Application): LeakTracing {
//        return LeakCanaryTracing(application)
//    }
//
//}
